deployment/src/main/java/io/quarkiverse/fury/deployment/FuryProcessor.java

```diff
@@ -1,14 +1,63 @@
 package io.quarkiverse.fury.deployment;
 
+import java.util.List;
+
+import org.jboss.jandex.AnnotationTarget;
+import org.jboss.jandex.DotName;
+
+import io.quarkiverse.fury.FuryBuildTimeConfig;
+import io.quarkiverse.fury.FuryProducer;
+import io.quarkiverse.fury.FuryRecorder;
+import io.quarkiverse.fury.FurySerialization;
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.arc.deployment.BeanContainerBuildItem;
+import io.quarkus.arc.deployment.UnremovableBeanBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
 import io.quarkus.deployment.builditem.FeatureBuildItem;
 
 class FuryProcessor {
 
     private static final String FEATURE = "fury";
+    private static final DotName FURY_SERIALIZATION = DotName.createSimple(FurySerialization.class.getName());
 
     @BuildStep
     FeatureBuildItem feature() {
         return new FeatureBuildItem(FEATURE);
     }
+
+    @BuildStep
+    public void findSerializableClasses(CombinedIndexBuildItem combinedIndex,
+            BuildProducer<FurySerializerBuildItem> pojoProducer) {
+        combinedIndex.getIndex().getAnnotations(FURY_SERIALIZATION)
+                .stream()
+                .filter(annotation -> annotation.target().kind() == AnnotationTarget.Kind.CLASS)
+                .forEach(i -> {
+                    pojoProducer.produce(new FurySerializerBuildItem(i.target().asClass()));
+                });
+    }
+
+    @BuildStep
+    void unremovableBeans(
+            BuildProducer<AdditionalBeanBuildItem> beanProducer,
+            BuildProducer<UnremovableBeanBuildItem> unremovableBeans) {
+        beanProducer.produce(AdditionalBeanBuildItem.unremovableOf(FuryProducer.class));
+    }
+
+    @BuildStep
+    @Record(ExecutionTime.STATIC_INIT)
```

Comment:
We need one place to invoke registration to generate fury serializer at build time

Reply:
Yeah, it is in `FuryRecorder`

Reply:
`ExecutionTime.STATIC_INIT` means when building in the native mode, this method is recordered and running at build time.

Reply:
I see ,thanks